drifter_processing.py

```diff
@@ -0,0 +1,454 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Nov  7 16:00:10 2019
+
+@author: strausz
+"""
+
+import pandas as pd
+import cartopy.crs as ccrs
+import matplotlib.pyplot as plt
+#import cartopy.feature as cfeature
+import cmocean
+#for calculating distance
+from haversine import haversine
+import argparse
+from datetime import datetime
+import numpy as np
+import re
+from erddapy import ERDDAP
+import mysql.connector
+
+
+
+parser = argparse.ArgumentParser(description='Plot drifter track on map')
+parser.add_argument('-if','--infile', nargs=1, type=str, 
+                    help='full path to input file')
+parser.add_argument('-p', '--plot', type=str, 
+                    help="make plot of 'sst', 'strain', or 'speed' ")
+parser.add_argument('-f', '--file', action="store_true",
+                    help="output csv file of data")
+parser.add_argument('-i', '--ice', action="store_true",
+                    help="add ice concentration as last field and output file")
+parser.add_argument('-ph', '--phyllis', action="store_true",
+                    help="output format for phyllis friendly processing")
+parser.add_argument('-d', '--date', action="store_true",
+                    help="add occasional date to track")
+parser.add_argument('-e', '--erddap', nargs='+',
+                    help="get directly from akutan erddap server, requires argos id followed desired years")
+parser.add_argument('-H', '--hour', action="store_true",
+                    help="resample all data to even hour and interpolate")
+parser.add_argument('-s', '--speed', action="store_true",
+                    help="add speed column")
+parser.add_argument('-V', '--vecdis', action="store_true",
+                    help="import a vecdis file as input")
+parser.add_argument('-l', '--legacy', nargs='?',
+                    help="file has legacy format from ecofoci website, if file contains ice concentraion, add 'i'")
+parser.add_argument('-c', '--cut', nargs='*',
+                    type=lambda x: datetime.strptime(x, "%Y-%m-%dT%H:%M:%S"),
+                    help="date span in format '2019-01-01T00:00:00 2019-01-01T00:00:01' for example")
+parser.add_argument('-de', '--despike', action="store_true",
+                     help="Do some simple despiking of sst")
+args=parser.parse_args()
+
+
+#the following is info needed for adding the ice concentration 
```

Comment:
The hard coded paths to content with this archive/project should be converted to relative paths (no reference to your own system)

hard coded paths to content outside of this project could be set in a config file that is read in or in the command line entries as a default value but shouldn't be hidden in the body of the code if possible